src/fscacher/cache.py

```diff
@@ -122,7 +121,11 @@ def fingerprinter(*args, **kwargs):
             bound = sig.bind(*args, **kwargs)
             bound.apply_defaults()
             path_orig = bound.arguments[path_arg]
-            path = op.realpath(path_orig)
+            try:
+                path = op.realpath(path_orig)
+            except TypeError:
+                lgr.debug("Argument is not a path-like object; bypassing cache")
```

Comment:
```suggestion
                lgr.debug("Calling %s directly since argument is not a path-like object", f)
```
for consistency with other instances (below) where we bypass the cache.